monai/metrics/meandice.py

```diff
@@ -66,29 +69,29 @@ def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor
             y_pred: input data to compute, typical segmentation model output.
                 It must be one-hot format and first dim is batch, example shape: [16, 3, 32, 32]. The values
                 should be binarized.
-            y: ground truth to compute mean dice metric. It must be one-hot format and first dim is batch.
-                The values should be binarized.
+            y: ground truth to compute mean Dice metric. `y` can be single-channel class indices or
+                in the one-hot format.
 
         Raises:
-            ValueError: when `y` is not a binarized tensor.
             ValueError: when `y_pred` has less than three dimensions.
         """
-        is_binary_tensor(y_pred, "y_pred")
-        is_binary_tensor(y, "y")
-
         dims = y_pred.ndimension()
         if dims < 3:
             raise ValueError(f"y_pred should have at least 3 dimensions (batch, channel, spatial), got {dims}.")
         # compute dice (BxC) for each channel for each batch
-        return compute_dice(
-            y_pred=y_pred, y=y, include_background=self.include_background, ignore_empty=self.ignore_empty
-        )
+        return DiceHelper(  # type: ignore
```

Comment:
I would always advise against instantiating a class just to call a method like this, it's like a poltergeist. This instance of `DiceHelper` should be a member value which is used here instead.